openedx_blockstore_relay/transfer_data.py

```diff
@@ -103,42 +139,27 @@ def block_key(self):
         """
         return self.block.location
 
+    @property
+    def block_files_prefix(self):
+        """
+        For a given unit, all non-public files should be in this "folder"
+        within the bundle, e.g. /unit-first_unit/
+        """
+        return '/' + self.block_key.block_type + '-' + self.block_key.block_id + '/'
+
     def upload_olx(self, block):
         """
-        Upload the OLX for the given block definition, recursing through children.
+        Upload the OLX for the given XBlock/XModule and its children
         """
         block_key = block.location
         if block_key in self._block_olx_ok:
             # Prevent infinite recursion: don't re-create OLX for blocks that we've already done.
             return None
 
-        block_type = block_key.block_type
-        if block_type in ('course', 'chapter', 'sequential', 'vertical'):
-            # Once we have a runtime, we can call `definition_to_xml` for these blocks too.
-            olx_node = ElementTree.Element(block_type)
-        else:
-            try:
-                filesystem = WrapFS(MemoryFS())
-                olx_node = block.definition_to_xml(filesystem)
-                self._upload_related_files(filesystem)
-            except (NotImplementedError, AttributeError):
-                LOG.warning("Block type not supported, skipping (%s)", block_key)
-                self._block_olx_ok[block_key] = False
-                return None
-            except ResourceNotFound as exc:
-                LOG.error("Error fetching block OLX (%s): %s", block_key, exc)
-                self._block_olx_ok[block_key] = False
-                return None
-
-        for child in block.get_children():
-            LOG.debug('Add child %s', child.location)
-            self.upload_olx(child)
-            if self._block_olx_ok[child.location]:
-                # If we've successfully processed this child's OLX,
-                # include a reference to the child in the parent block.
-                child_ref = ElementTree.Element(child.location.block_type)
-                child_ref.set('url_name', child.url_name)
-                olx_node.append(child_ref)
+        olx_node = Element("unknown_root", nsmap=XML_NAMESPACES)
```

Comment:
What is the significance of "unknown_root"?

Reply:
This code is copied from the `XBlock` code repo. The name of the node doesn't matter since the XBlock will rename the node when it adds its data to it.

Reply:
Ah okay! Maybe put in a note for future readers of this code?